src/sparql/QueryObjectBuilder.ts

```diff
@@ -52,6 +54,46 @@ export default class QueryObjectBuilder {
 			},
 		];
 
+		if ( queryRepresentation.condition.propertyValueRelation === PropertyValueRelation.NotMatching ) {
```

Comment:
This is repeating a lot of query building logic from the above. Can't it have some cleaner way to do it?

Reply:
Hi, I am aware of this, will do in next iteration.